server/src/main/java/com/saralein/server/Application.java

```diff
@@ -1,48 +1,50 @@
 package com.saralein.server;
 
-import com.saralein.server.controller.ErrorController;
-import com.saralein.server.controller.DirectoryController;
-import com.saralein.server.controller.FileController;
-import com.saralein.server.logger.Logger;
-import com.saralein.server.request.RequestParser;
-import com.saralein.server.response.ResponseSerializer;
-import com.saralein.server.router.Routes;
+import com.saralein.server.middleware.Callable;
+import com.saralein.server.middleware.Middleware;
+import com.saralein.server.middleware.StaticMiddleware;
+import com.saralein.server.request.Request;
+import com.saralein.server.response.Response;
 import com.saralein.server.router.Router;
+import com.saralein.server.router.Routes;
 
 import java.nio.file.Path;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 
 public class Application {
-    private final Logger logger;
-    private Routes routes;
+    private final Callable callable;
 
-    public Application(Logger logger) {
-        this.logger = logger;
-        this.routes = new Routes();
+    public Application(Callable callable) {
+        this.callable = callable;
     }
 
-    public Application config(Routes routes) {
-        this.routes = routes;
-        return this;
+    public Response call(Request request) {
+        return callable.call(request);
     }
 
-    public void start(int port, Path root) {
-        Runtime runtime = Runtime.getRuntime();
-        FileHelper fileHelper = new FileHelper(root);
+    public static class Builder {
```

Comment:
I am curious, what does this Builder do for us? Does it make something easy? hard?